### PR TITLE
Issue-497: Headers in ES should be decoded

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: mpt/app/dependency-reduced-pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <parent>
```
Comment (chibenwa) [inline mpt/app/dependency-reduced-pom.xml]: Irelevant file

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature [hunk 2]
```diff
@@ -7,74 +7,73 @@ Feature: Download endpoint
...
-    When "userb@domain.tld" downloads "a1"
-    Then the user should receive a not found response
-
-  @Ignore
-  Scenario: An authenticated user should have access to a shared attachment
-    Given "usera@domain.tld" shares its mailbox "INBOX" with "userb@domain.tld"
-    And "userb@domain.tld" is connected
-    When "userb@domain.tld" downloads "a1"
-    Then the user should be authorized
+#  Scenario: An authenticated user should initiate the access to the download endpoint
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: Looks like you don't really need this commit

**Comment by chibenwa** [comment]
Remove the "Test" commit, create a JIRA issue on APACHE JIRA, rename your commit and you have my +1

File: mpt/app/dependency-reduced-pom.xml (hunk 1 above)
Comment (quynhn) [inline mpt/app/dependency-reduced-pom.xml]: I removed the wrong commit :(

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature (hunk 2 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: I removed the wrong commit :(

**Comment by quynhn** [comment]
It's done with: remove wrong commit, create ticket on Jira and committed

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java [hunk 3]
```diff
@@ -62,7 +62,16 @@ public void displayNamesShouldBeRetreived() {
...
+    public void displayNamesShouldBeRetrievedWhenEncodedWord() {
+        HeaderCollection headerCollection = HeaderCollection.builder().add(new FieldImpl("From", "=?UTF-8?B?RnLDqWTDqXJpYyBNQVJUSU4=?= <fmartin@linagora.com>, Graham CROSMARIE <gcrosmarie@linagora.com>")).build();
+        assertThat(headerCollection.getFromAddressSet())
+            .containsOnly(new EMailer("Frédéric MARTIN", "fmartin@linagora.com"),
+                new EMailer("Graham CROSMARIE", "gcrosmarie@linagora.com"));
+
+    }
+
+    @Test
+    public void addressWithTwoDisplayNamesOnTheSameFieldShouldBeRetreived1() {
```
Comment (aduprat) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: Remove the `1` at the end of the method name

Comment (quynhn) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: Yes, I removed

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java [hunk 4]
```diff
@@ -62,7 +62,16 @@ public void displayNamesShouldBeRetreived() {
     }
 
     @Test
-    public void addressWithTwoDisplayNamesOnTheSameFieldShouldBeRetreived() {
+    public void displayNamesShouldBeRetrievedWhenEncodedWord() {
+        HeaderCollection headerCollection = HeaderCollection.builder().add(new FieldImpl("From", "=?UTF-8?B?RnLDqWTDqXJpYyBNQVJUSU4=?= <fmartin@linagora.com>, Graham CROSMARIE <gcrosmarie@linagora.com>")).build();
+        assertThat(headerCollection.getFromAddressSet())
+            .containsOnly(new EMailer("Frédéric MARTIN", "fmartin@linagora.com"),
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: you don't really care for that : you just want to make sure "Frédéric" is ok, so you can do `.extract(Emailer::name).contains("Frédéric MARTIN")` (it probably doesn't compile, but I guess you got the point).

Comment (quynhn) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: Yes, then I changed to
```
assertThat(headerCollection.getFromAddressSet())
            .extracting(EMailer::getName)
            .contains("Frédéric MARTIN");
```

**Pull request merged**